airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteFileOffsetBackingStore.java

```diff
@@ -59,14 +60,40 @@ public Map<String, String> read() {
   public void persist(final JsonNode cdcState) {
     final Map<String, String> mapAsString =
         cdcState != null ? Jsons.object(cdcState, Map.class) : Collections.emptyMap();
-    final Map<ByteBuffer, ByteBuffer> mappedAsStrings = mapAsString.entrySet().stream().collect(Collectors.toMap(
+    final Map<String, String> updatedMap = downgradeStateForDebezium1_9(mapAsString);
+
+    final Map<ByteBuffer, ByteBuffer> mappedAsStrings = updatedMap.entrySet().stream().collect(Collectors.toMap(
         e -> stringToByteBuffer(e.getKey()),
         e -> stringToByteBuffer(e.getValue())));
 
     FileUtils.deleteQuietly(offsetFilePath.toFile());
     save(mappedAsStrings);
   }
 
+  // This is to allow us to release Debezium 2.1.2 safely so that in case of a roll back we are able
+  // to convert the new state from Debezium 2.1.2 back to the format of Debezium 1.9.6
+  private Map<String, String> downgradeStateForDebezium1_9(final Map<String, String> mapAsString) {
```

Comment:
Should we only downgrade state if it 2.1 (i.e. 1.9 -> 1.9 should be a no-op)

Reply:
Yup thats what its doing. It has the logic in place for that. 
```
if (i == 0 && i1 == key.length() - 1) {
        // The state is Debezium 1.9 compatible. No need to change anything.
        return mapAsString;
      }
```

Reply:
I need to extract the keys to make that decision

Reply:
Wow, sorry missed this. Could you (maybe in the PR description) give an example of what the 1.9 state looks like vs 2.1 state? Just thinking of a case where it's easy for someone oncall (me..) to go back and easily determine whether a state is assc with 1.9 or 2.1

Reply:
Done